src/clj/rems/api/applications_v2.clj

```diff
@@ -0,0 +1,813 @@
+                                                                         (get-in license [:license/text lang]))}]))})
+                    (:application/licenses application))
+     :items (map (fn [field]
+                   {:id (:field/id field)
+                    :type (name (:field/type field))
+                    :optional (:field/optional field)
+                    :options (:field/options field)
+                    :maxlength (:field/max-length field)
+                    :value (:field/value field)
+                    :previous-value nil ; TODO
+                    :localizations (into {} (for [lang (set (concat (keys (:field/title field))
+                                                                    (keys (:field/placeholder field))))]
+                                              [lang {:title (get-in field [:field/title lang])
+                                                     :inputprompt (get-in field [:field/placeholder lang])}]))})
+                 (:form/fields application))}))
+
+(defn api-get-application-v1 [user-id application-id]
+  (when-let [v2 (api-get-application-v2 user-id application-id)]
+    (transform-v2-to-v1 v2 user-id)))
+
+;;; v2 API, listing all applications
+
+(defn- apply-event [applications event]
+  (if-let [app-id (:application/id event)] ; old style events don't have :application/id
+    (update applications app-id update-application-view event)
+    applications))
+
+(defn- apply-events [applications events]
+  (reduce apply-event applications events))
+
+(defn- reset-state [_]
+  {:last-processed-event-id 0
+   :applications {}})
+
+(def projection-state (agent (reset-state nil)
+                             :error-handler (fn [_agent exception]
+                                              (log/error exception "Updating projection failed"))))
+(comment
+  @projection-state
+  (send projection-state reset-state))
+
+(defn update-projection [state]
+  (let [from-id (:last-processed-event-id state)
+        batch-size 1000
+        events (applications/get-dynamic-application-events-since from-id batch-size)
+        until-id (:event/id (last events))]
+    (if (empty? events)
+      state
+      (do
+        (log/info "Updating projection from" from-id "until" until-id)
+        (when (= batch-size (count events))
+          ;; there may be more events, so handle the next batch immediately after the current one
+          (send-off projection-state update-projection))
+        (assoc state
+               :last-processed-event-id until-id
+               :applications (apply-events (:applications state) events))))))
+
+(defstate projection-updater
+  :start (future (while (not (Thread/interrupted))
+                   (send-off projection-state update-projection)
+                   (Thread/sleep (-> 60 time/seconds time/in-millis))))
+  :stop (future-cancel projection-updater))
```

Comment:
I'd like a short comment about this architecture. I see agents + futures + mount. How does it all fit together and what is the goal?